### PR TITLE
Current day

### PR DESCRIPTION
**Pull request by aflorczak**
Fixed OrderRestApi with all services and CarRestAPi with all services.

### CODE REVIEW

File: src/main/java/com/carflow/backend/domains/cars/services/CarService.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import com.carflow.backend.domains.cars.interfaces.out.CarStorage;
 import com.carflow.backend.exceptions.ObjectNotFoundException;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/domains/cars/services/CarService.java:7]: unused import

File: src/main/java/com/carflow/backend/infrastructure/boundaries/in/api/v_0_0_1/cars/controllers/CarRestApi.java [hunk 2]
```diff
@@ -24,23 +24,14 @@ public CarRestApi(CarService carService, CarConverter carConverter) {
     }
 
     @PostMapping("/cars")
-    public Car createNewCar(@RequestBody CarDto carRequest) {
+    public CarDto createNewCar(@RequestBody CarDto carRequest) {
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/infrastructure/boundaries/in/api/v_0_0_1/cars/controllers/CarRestApi.java:27]: 👍

File: src/main/java/com/carflow/backend/infrastructure/boundaries/in/api/v_0_0_1/cars/controllers/CarRestApi.java [hunk 3]
```diff
@@ -56,6 +47,11 @@ public CarDto updateCarById(@PathVariable String id, @RequestBody CarDto carDto)
         return carConverter.convertCarToCarDto(carResponse);
     }
 
+    @PatchMapping("/cars/{id}/archive")
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/infrastructure/boundaries/in/api/v_0_0_1/cars/controllers/CarRestApi.java:50]: Patch should be used for partial object modification - you are trying to perform "archive" operation - use POST

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/entity/CarEntity.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
+package com.carflow.backend.infrastructure.boundaries.out.cars.entity;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+
+@Entity
+public class CarEntity {
+    // System properties
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/entity/CarEntity.java:9]: remove confusing comment

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java [hunk 5]
```diff
@@ -69,6 +54,19 @@ public Car updateCarById(String id, Car updatedCar) throws ObjectNotFoundExcepti
         }
     }
 
+    @Override
+    public Car moveToArchive(String id) throws ObjectNotFoundException {
+        if (cars.containsKey(id)) {
+            Car car = cars.get(id);
+            Car carArchive = cars.get(id);
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java:61]: car and carArchive are exactly the same object

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java [hunk 6]
```diff
@@ -69,6 +54,19 @@ public Car updateCarById(String id, Car updatedCar) throws ObjectNotFoundExcepti
...
+            carArchive.setArchive(true);
+            cars.replace(id, car, carArchive);
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java:63]: replace is not needed - you are modifying the exact same object in memory

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java [hunk 7]
```diff
@@ -69,6 +54,19 @@ public Car updateCarById(String id, Car updatedCar) throws ObjectNotFoundExcepti
...
+            return cars.get(id);
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java:64]: just return the object you already have - no need to again look it up in the map

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        return carRepository.findAll().stream().map(carEntityConverter::convertCarEntityToCar).toList();
+    }
+
+    @Override
+    public Car getCarById(String id) throws ObjectNotFoundException {
+        Optional<CarEntity> car = carRepository.findById(id);
+        if (car.isPresent()) {
+            return carEntityConverter.convertCarEntityToCar(car.get());
+        } else {
+            throw new ObjectNotFoundException("Obiekt nie istnieje");
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java:43]: english

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Override
+    public Car moveToArchive(String id) throws ObjectNotFoundException {
+        Optional<CarEntity> car = carRepository.findById(id);
+        if (car.isPresent()) {
+            CarEntity updatedCar = car.get();
+            updatedCar.setArchive(true);
+            return carEntityConverter.convertCarEntityToCar(carRepository.save(updatedCar));
+        } else {
+            throw new ObjectNotFoundException("Obiekt nie istnieje");
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java:61]: en

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+
+    @Override
+    public void deleteCarById(String id) throws ObjectNotFoundException {
+        Optional<CarEntity> car = carRepository.findById(id);
+        if (car.isPresent()) {
+            carRepository.deleteById(id);
+        } else {
+            throw new ObjectNotFoundException("Obiekt nie istnieje");
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java:71]: en

File: src/main/java/com/carflow/backend/runner/DataRunner.java [hunk 11]
```diff
@@ -9,15 +11,19 @@
...
-    public DataRunner(OrderRepository repo) {
-        this.repo = repo;
+    public DataRunner(OrderRepository orderRepository, CarRepository carRepository) {
+        this.orderRepository = orderRepository;
+        this.carRepository = carRepository;
     }
 
     @Override
     public void run(ApplicationArguments args) throws Exception {
+        System.out.println("Runner");
```
Comment (dawidowicz) [inline src/main/java/com/carflow/backend/runner/DataRunner.java:24]: remember to prepare proper logging using logger like log4j2 later

File: src/main/resources/application.properties [hunk 12]
```diff
@@ -1,3 +1,4 @@
 spring.h2.console.enabled=true
-spring.datasource.url=jdbc:h2:~/test
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.url=jdbc:h2:~/test;AUTO_SERVER=TRUE
 spring.jpa.hibernate.ddl-auto=create
```
Comment (dawidowicz) [inline src/main/resources/application.properties:4]: missing line end in file - this is shown but the red symbol in github. convention is that each file should have an empty last line (end with \n or similar depending on system)

**Review by dawidowicz — CHANGES_REQUESTED** [review]

File: src/main/java/com/carflow/backend/domains/cars/services/CarService.java (hunk 1 above)
Comment (aflorczak) [reply src/main/java/com/carflow/backend/domains/cars/services/CarService.java:7]: ok

File: src/main/java/com/carflow/backend/infrastructure/boundaries/in/api/v_0_0_1/cars/controllers/CarRestApi.java (hunk 2 above)
Comment (aflorczak) [reply src/main/java/com/carflow/backend/infrastructure/boundaries/in/api/v_0_0_1/cars/controllers/CarRestApi.java:27]: ok

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java (hunk 9 above)
Comment (aflorczak) [reply src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java]: ok

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java (hunk 10 above)
Comment (aflorczak) [reply src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java]: ok

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java (hunk 8 above)
Comment (aflorczak) [reply src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/SQLCarStorage.java]: ok

File: src/main/resources/application.properties (hunk 12 above)
Comment (aflorczak) [reply src/main/resources/application.properties]: ok

File: src/main/java/com/carflow/backend/runner/DataRunner.java (hunk 11 above)
Comment (aflorczak) [reply src/main/java/com/carflow/backend/runner/DataRunner.java:24]: https://github.com/aflorczak/com.carflow.backend/issues/3

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java (hunk 5 above)
Comment (aflorczak) [reply src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java:61]: ok

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java (hunk 6 above)
Comment (aflorczak) [reply src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java:63]: ok

File: src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java (hunk 7 above)
Comment (aflorczak) [reply src/main/java/com/carflow/backend/infrastructure/boundaries/out/cars/implementations/CarStorageMemory.java:64]: ok

File: src/main/java/com/carflow/backend/infrastructure/boundaries/in/api/v_0_0_1/cars/controllers/CarRestApi.java (hunk 3 above)
Comment (aflorczak) [reply src/main/java/com/carflow/backend/infrastructure/boundaries/in/api/v_0_0_1/cars/controllers/CarRestApi.java]: ok

**Review by dawidowicz — APPROVED** [review]

**Pull request merged**